R/RunWorkflow.R

```diff
@@ -99,6 +199,27 @@ RunWorkflow <- function(
     } else {
       cli::cli_h2("Returning results from final step: {typeof(lWorkflow$lResult)} of length {length(lWorkflow$lResult)}`.")
     }
+
+    # Save data with configuration object.
+    if (!is.null(lConfig)) {
+      if (
+        exists('SaveData', lConfig) &&
+        is.function(lConfig$SaveData) &&
+        all(c('lWorkflow', 'lConfig') %in% names(formals(lConfig$SaveData)))
+      ) {
+        cli::cli_h3('Saving data with `lConfig$SaveData`.')
```

Comment:
@zdz2101 FYI- I'm going to merge this in before the logging PR gets merged, so a few new `cli::` calls will need to be changed over in that PR once this gets merged to dev